src/main/java/sirius/db/mixing/types/SafeMap.java

```diff
@@ -55,13 +55,19 @@
 
     /**
      * Puts the given key and value into the map.
+     * <br>
+     * Keys with null values will be removed.
      *
      * @param key   the key used to store the value
      * @param value the value to store
      * @return the map itself for fluent method calls
      */
     public SafeMap<K, V> put(@Nonnull K key, V value) {
-        modify().put(key, value);
+        if (value != null) {
```

Comment:
shouldn't we only overwrite this in MultiLanguageString to avoid unwanted side effects?